aikau/src/main/resources/alfresco/core/CoreWidgetProcessing.js

```diff
@@ -51,6 +51,110 @@ define(["dojo/_base/declare",
       cssRequirements: [{cssFile:"./css/CoreWidgetProcessing.css"}],
 
       /**
+       * This string is used to identify locations of counts of widgets that are being processed.
+       *
+       * @instance
+       * @type {string}
+       * @default
+       * @since 1.0.36
+       */
+      _countDownLocationPrefix: "_processingCountDown",
+
+      /**
+       * This string is used to identify locations where either processed widgets can be referenced. This location
+       * will either be populated with an array of widgets or with a promise that will be resolved once all the
+       * widgets have been created. This should not be set or configured.
+       *
+       * @instance
+       * @type {string}
+       * @default
+       * @since 1.0.36
+       */
+      _processedWidgetsLocationPrefix: "_processedWidgets",
+
+      /**
+       * This string is used to identify locations of arrays where widgets that are being created will be stored. 
+       * THis should not be set or configured.
+       *
+       * @instance
+       * @type {string}
+       * @default
+       * @since 1.0.36
+       */
+      _processingWidgetsLocationPrefix: "_processingWidgets",
+
+      /**
+       * Used to keep track of all the widgets created as a result of a call to the 
+       * [processWidgets]{@link module:alfresco/core/Core#processWidgets} function. This should not be referenced directly,
+       * instead the [getProcessedWidgets]{@link module:alfresco/core/Core#getProcessedWidgets} should be called.
+       *
+       * @instance
+       * @type {Array}
+       * @default
+       */
+      _processedWidgets: null,
+
+      /**
+       * As of 1.0.36 this is no longer used and it should not be referenced
+       *
+       * @instance
+       * @type {number}
+       * @default
+       * @deprecated Since 1.0.36
+       */
+      _processedWidgetCountdown: null,
+      
+      /**
+       * This function can be used to retrieve the widgets that have been processed by a call to 
+       * [processWidgets]{@link module:alfresco/core/CoreWidgetProcessing#processWidgets}. It return a promise
+       * of the widgets that will be created an as such calls to this function should be protected through
+       * the use of dojo/when. It is not recommended to use this function, instead it is better to extend
+       * the [allWidgetsProcessed]{@link module:alfresco/core/CoreWidgetProcessing#allWidgetsProcessed} function
+       * as this will only be called when widget processing has completed and is passed an argument of the
+       * widgets that were created.
+       * 
+       * @instance
+       * @param  {string} processWidgetsId The ID mapped to the original processWidgets call
+       * @return {object[]|promise} The array of processed widgets or a promise of them
+       * @since 1.0.36
+       */
+      getProcessedWidgets: function alfresco_core_CoreWidgetProcessing__getProcessedWidgets(processWidgetsId) {
+         var r;
```

Comment:
widgets? processedWidgets? returnWidgets? single-letters are just really bad for variable names :)